File: namsor/java/src/main/java/oseg/namsor_examples/ChineseNameMatchBatchExample.java
package oseg.namsor_examples;

import app.namsor.client.ApiException;
import app.namsor.client.Configuration;
import app.namsor.client.api.*;
import app.namsor.client.auth.*;
import app.namsor.client.JSON;
import app.namsor.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class ChineseNameMatchBatchExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("api_key")).setApiKey("YOUR_API_KEY");

        var personalNames1Name1 = new FirstLastNameIn();
        personalNames1Name1.id("e630dda5-13b3-42c5-8f1d-648aa8a21c42");
        personalNames1Name1.firstName("Hong");
        personalNames1Name1.lastName("Yu");

        var personalNames1Name2 = new PersonalNameIn();
        personalNames1Name2.id("e630dda5-13b3-42c5-8f1d-648aa8a21c43");
        personalNames1Name2.name("喻红");

        var personalNames1 = new MatchPersonalFirstLastNameIn();
        personalNames1.id("e630dda5-13b3-42c5-8f1d-648aa8a21c41");
        personalNames1.name1(personalNames1Name1);
        personalNames1.name2(personalNames1Name2);

        var personalNames = new ArrayList<MatchPersonalFirstLastNameIn>(List.of (
            personalNames1
        ));

        var batchMatchPersonalFirstLastNameIn = new BatchMatchPersonalFirstLastNameIn();
        batchMatchPersonalFirstLastNameIn.personalNames(personalNames);

        try
        {
            var response = new ChineseApi(config).chineseNameMatchBatch(
                batchMatchPersonalFirstLastNameIn
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ChineseApi#chineseNameMatchBatch");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: namsor/kotlin/src/main/kotlin/oseg/namsor_examples/ChineseNameMatchBatchExample.kt
package oseg.namsor_examples

import app.namsor.client.infrastructure.*
import app.namsor.client.apis.*
import app.namsor.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class ChineseNameMatchBatchExample
{
    fun chineseNameMatchBatch()
    {
        ApiClient.apiKey["X-API-KEY"] = "YOUR_API_KEY"

        val personalNames1Name1 = FirstLastNameIn(
            id = "e630dda5-13b3-42c5-8f1d-648aa8a21c42",
            firstName = "Hong",
            lastName = "Yu",
        )

        val personalNames1Name2 = PersonalNameIn(
            id = "e630dda5-13b3-42c5-8f1d-648aa8a21c43",
            name = "喻红",
        )

        val personalNames1 = MatchPersonalFirstLastNameIn(
            id = "e630dda5-13b3-42c5-8f1d-648aa8a21c41",
            name1 = personalNames1Name1,
            name2 = personalNames1Name2,
        )

        val personalNames = arrayListOf<MatchPersonalFirstLastNameIn>(
            personalNames1,
        )

        val batchMatchPersonalFirstLastNameIn = BatchMatchPersonalFirstLastNameIn(
            personalNames = personalNames,
        )

        try
        {
            val response = ChineseApi().chineseNameMatchBatch(
                batchMatchPersonalFirstLastNameIn = batchMatchPersonalFirstLastNameIn,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling ChineseApi#chineseNameMatchBatch")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ChineseApi#chineseNameMatchBatch")
            e.printStackTrace()
        }
    }
}
